models/autoencoder.py

```diff
@@ -0,0 +1,52 @@
+import torch
+import torch.utils.data
+from torch import nn, optim
+from torch.nn import functional as F
+
+from data_loader import get_data_loaders
+from models.layers import CNN_Decoder, CNN_Encoder
+
+
+class Network(nn.Module):
+    def __init__(self, args):
+        super(Network, self).__init__()
+        self.args = args
+        output_size = args.model.embedding_size
+        if args.dataset.name == "CIFAR10":
+            self.encoder = CNN_Encoder(output_size, input_size=(3, 32, 32))
+            self.decoder = CNN_Decoder(args.model.embedding_size, input_size=(3, 32, 32))
+        else:
+            self.encoder = CNN_Encoder(output_size)
+            self.decoder = CNN_Decoder(args.model.embedding_size)
```

Comment:
I think you can use the `input_size` as an argument explicitly, instead of inferring it from dataset name (model shouldn't care about dataset name).